### PR TITLE
Rollup some PRs in the queue

### PR DESCRIPTION
**Pull request by Manishearth**
(still testing locally)

### CODE REVIEW

**Comment by steveklabnik** [comment]
I hope this works for you. I've been trying for a few days and it passes locally but fails on Windows...

**Comment by Manishearth** [comment]
Hmm. I removed the cloned PR since it seemed to be causing repeated failures. If it's still an issue I'll look further. Thanks!

File: src/doc/trpl/ffi.md [hunk 1]
```diff
@@ -436,12 +436,19 @@ extern {
 
 fn main() {
     let prompt = CString::from_slice(b"[my-awesome-shell] $");
-    unsafe { rl_prompt = prompt.as_ptr(); }
-    // get a line, process it
-    unsafe { rl_prompt = ptr::null(); }
+    unsafe { 
+        rl_prompt = prompt.as_ptr();
+
+        println!("{}", rl_prompt);
```
Comment (steveklabnik) [inline src/doc/trpl/ffi.md]: :cry:

**Pull request merged**